### DISCUSSION TITLE
PDF File to base64

### DISCUSSION THREAD

**Discussion by almukhalafi** [spatie/laravel-pdf#140] [Q&A]
Hey everyone!

I hope you're all doing well. I'm currently working on a Laravel project and could really use some assistance from the community.

I'm encountering an issue with one of my jobs, SendWhatsAppInvoice, which is causing an error related to the type of PDF being passed. Here's the error message I'm facing:

`App\Jobs\SendWhatsAppInvoice::__construct(): Argument #2 ($pdf) must be of type Spatie\LaravelPdf\Facades\Pdf, Spatie\LaravelPdf\PdfBuilder given`

I've already attempted to resolve this by adjusting the type hinting in the constructor of the SendWhatsAppInvoice class, but I'm still running into issues.

Here's the relevant code snippet from the job:

```
use Spatie\LaravelPdf\Facades\Pdf;

// Job constructor
public function __construct(
    public readonly ?array $props,
    public readonly Pdf $pdf,
    public readonly Model $invoice,
    public readonly string $phone,
    public readonly WhatsAppSendDocumentContract $service
)
{
    // Constructor logic
}

use Spatie\LaravelPdf\Facades\Pdf;

// Dispatching the job
$pdf = Pdf::view('invoice.' . $data['invoice_view'], $data);

SendWhatsAppInvoice::dispatchSync(
    $whatApp->pivot->props_value,
    $pdf,
    $invoice,
    $phone,
    $service
);
```

I believe the issue might be with how I'm dispatching the job or how I'm using the PDF facade. I'm hoping someone in the community might have encountered a similar issue before and can offer some guidance or suggestions for troubleshooting.

Any help or advice would be greatly appreciated! Thank you in advance for your assistance.

Best regards,

**Comment by solomoreal** [comment]
you can remove the PDF type hint to fix the problem, like so:
public function __construct(
    public readonly ?array $props,
    public readonly $pdf,
    public readonly Model $invoice,
    public readonly string $phone,
    public readonly WhatsAppSendDocumentContract $service
)
